File: LeetCode/Medium/Group Anagrams/Java/groupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();
        List<List<String>> res = new ArrayList<>();
        for (String str : strs) {
            char[] c = str.toCharArray();
            Arrays.sort(c);
            String sb = "";
            for (int i=0; i<c.length; i++) sb += c[i];
            map.computeIfAbsent(sb, k -> new ArrayList<>()).add(str);
        }
        for (String s : map.keySet()) res.add(map.get(s));
        return res;
    }
}

File: LeetCode/neetcode 150/arrays and hashing/group_anagrams.cc
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> res;
        unordered_map<string, vector<string>> hash_map;
        for (string s : strs) {
            string temp = s;
            sort(temp.begin(), temp.end());
            hash_map[temp].push_back(s);
        }
        for (auto [k, v] : hash_map) {
            res.push_back(v);
        }
        return res;
    }
};
